common/src/com/thoughtworks/go/remote/work/ConsoleOutputTransmitter.java

```diff
@@ -86,10 +94,9 @@ public void flushToServer() {
             consoleAppender.append(result.toString());
         } catch (IOException e) {
             LOGGER.warn("Could not send console output to server", e);
-            //recreate buffer
             synchronized (buffer) {
                 sent.addAll(buffer);
-                buffer = new CircularFifoBuffer(10 * 1024);
+                buffer.clear();
```

Comment:
So, when the code was originally written, it used to be `buffer.clear()`. This was later changed to recreate the buffer as a part of a commit with a message "Fixed the performance problem". There were other changes made as a part of that commit, so I am not entirely sure if this was intentional or not. The change looks ok to me but I am still trying to think if anything will go wrong. Can you think of a reason?